src/apis/blockchain_api.ts

```diff
@@ -18,11 +23,30 @@ export class BlockchainAPI {
 
     private web3Utils: Web3Utils;
     private assert;
+    private contracts: ContractsAPI;
+    private isABIDecoderConfigured = false;
 
-    constructor(web3: Web3) {
+    constructor(web3: Web3, contracts: ContractsAPI) {
         this.web3Utils = new Web3Utils(web3);
         this.intervalManager = new IntervalManager();
         this.assert = new Assertions(web3);
+        this.contracts = contracts;
+    }
+
+    private async configureABIDecoder() {
+        if (this.isABIDecoderConfigured) {
+            return;
+        }
+        const debtKernel = await this.contracts.loadDebtKernelAsync();
```

Comment:
It is not necessary to asynchronously pull the contract wrapper in order to access the DebtKernel's abi -- we can simply import it directly from the `@dharmaprotocol/contracts` module.  Thus, we can synchronously configure the ABI decoder in the constructor instead.

Check out the DebtKernel wrapper for an example of this in action.